[VideoPlayback] Creates a video feed of multiple videos

GustavJaner opened the pull request:
# Description
- Query video info from DB
- Loads videos from S3
- Filters newest videos
- Possible to filter videos after user id

edgarlpassos commented on src/containers/Home.jsx:
```diff
@@ -1,11 +1,14 @@
     return (
       <Container className="home-container" fluid>
         <h1>Home Page</h1>
+        <VideoFeed numVideos="3" owner="36" />
```
Why are these values hardcoded?

GustavJaner replied:
Owner prop should not be passed in the Home.jsx. Only in for the Profile.jsx.

This is there for now, since otherwise the VideoFeed component won't load any videos due to the private file access as said

GustavJaner replied:
Changed the numVideos props to an optional route prop for Home.jsx instead.
owner should still be taken away when we have the private file access fixed

edgarlpassos commented on src/containers/Register.jsx:
```diff
@@ -0,0 +1,14 @@
+import React, { Component } from 'react';
+import { Container } from 'react-bootstrap';
+
+class Register extends Component {
```
There's already a registration page. You can delete this component.

GustavJaner replied:
Strange, this file was included in the commit not on purpose.

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react';
```
This file is present in the other PR you made, is this supposed to be the case, or did you accidentally include a commit you shouldn't have?

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+    const path = "/videos/" + this.props.id;
+
+    try {
+      const data = await API.get('videocloud', path);
```
You can use string interpolation and destructuring to make this prettier:
```
const { id } = this.props;
const data = await API.get('videocloud', `/videos/${id}`)';
```

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+      this.setState({ video: data });
+      this.s3loader();
+
+      console.log("state.video: ", this.state.video);
```
Delete the `console.log` statements.

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+
+      console.log("state.video: ", this.state.video);
+    } catch (error) {
+      console.log(error);
```
You can print the error, but use `console.error(..)`

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+      const url = await Storage.get(this.state.video.location, { level: 'private' })
+      this.setState({ S3url: url });
+
+      console.log("state.S3url: ", this.state.S3url);
```
Delete the `console.log` statements.

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+
+      console.log("state.S3url: ", this.state.S3url);
+    } catch (error) {
+      console.log(error);
```
Use `console.error()`

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+    }
+  }
+
+  loader(props) {
```
`props` is not used.

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+  };
+
+  async componentDidMount(props) {
+    const path = "/videos/" + this.props.id;
```
Same comment about string interpolation

edgarlpassos commented on src/containers/VideoFeed.jsx:
```diff
@@ -0,0 +1,94 @@
+    );
+
+    this.setState({ videos: filtered });
+    console.log("- filtered videos to load: ", this.state.videos);
```
Delete the `console.log` statements.

edgarlpassos commented on src/containers/VideoFeed.jsx:
```diff
@@ -0,0 +1,94 @@
+    return new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime();
+  }
+
+  filterQuery(videos, props) {
```
`props` is not used.

edgarlpassos commented on src/containers/Home.jsx:
```diff
@@ -1,11 +1,14 @@
 import React, { Component } from 'react';
 import { Container } from 'react-bootstrap';
 
+import VideoFeed from './VideoFeed';
```
If we're using `VideoFeed` inside `Home`, then `VideoFeed` should be in `src/components/`

edgarlpassos commented on src/containers/VideoFeed.jsx:
```diff
@@ -0,0 +1,94 @@
+   };
+
+  async componentDidMount(props) {
+    this.debugger();
```
Should this be here?

edgarlpassos commented on src/containers/VideoFeed.jsx:
```diff
@@ -0,0 +1,94 @@
+     owner: null, // if not sent as prop: loads videos from all users
+   };
+
+  async componentDidMount(props) {
```
`props` is not used.

edgarlpassos commented on src/containers/VideoFeed.jsx:
```diff
@@ -0,0 +1,94 @@
+      );
+    }
+
+    // Filters the newest videos
```
This comment suggests that you can extract the lines below into a function: `filterVideos()`.

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+  }
+
+  async s3loader(props) {
+    // Storage.configure({ level: 'private' });
```
Delete this commented line

edgarlpassos commented on src/containers/VideoItem.jsx:
```diff
@@ -0,0 +1,72 @@
+    }
+  }
+
+  async s3loader(props) {
```
`props` is not used.

GustavJaner committed:
[VideoPlayback] Creates a video feed of multiple videos
- Query video info from DB
- Loads videos from S3
- Set how many videos to render on Home page by the optional route prop: path=/:numVideos?
- Filters newest videos
- Possible to filter videos after user id

edgarlpassos reviewed (approved):
LGTM
